src/main/java/com/github/lordrex34/config/ConfigManager.java

```diff
@@ -57,11 +49,11 @@
 {
 	private static final Logger LOGGER = LoggerFactory.getLogger(ConfigManager.class);
 	
-	/** A simple {@link AtomicBoolean} that indicates reloading process. */
-	private final AtomicBoolean _reloading = new AtomicBoolean(false);
+	/** Contains all the registered {@link ConfigClassInfo}s. */
+	private final Set<ConfigClassInfo> _configRegistry = new HashSet<>();
 	
-	/** Properties registry, that is used for misplaced configuration indication. */
-	private final Map<String, Map<Path, Set<String>>> _propertiesRegistry = new TreeMap<>();
+	/** A simple {@link AtomicBoolean} that indicates reloading process. */
+	private static final AtomicBoolean RELOADING = new AtomicBoolean(false);
```

Comment:
what? static? why? you can have multiple config managers, and they are all bound by this? 

(not to mention this isn't proper synchronization I talked about :D I mean what should be handled is reloading from two threads at the same time :D )

Reply:
Done.